op/framework.py

```diff
@@ -4,6 +4,7 @@
 import types
 import sqlite3
 import collections
+import collections.abc
```

Comment:
It generates the following warning:

```
DeprecationWarning: Using or importing the ABCs from 'collections' instead of from 'collections.abc' is deprecated since Python 3.3,and in 3.9 it will stop working
```

We also agreed that we were only going to support 3.6+ IIRC

Reply:
Given that Xenial is still LTS and has 3.5.2 do we want to support back to 3.5? Otherwise I'm happy with Bionic+. Especially as "new charms probably won't target old LTS".